src/views/createprojectmodal/components/SelectClusterUDN.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { ResourceIcon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { DropdownItem, FormGroup } from '@patternfly/react-core';
+import InlineLoading from '@utils/components/Loading/InlineLoading';
+import Select from '@utils/components/Select/Select';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { ClusterUserDefinedNetworkModelGroupVersionKind } from '@utils/models';
+import { getName } from '@utils/resources/shared';
+import { ClusterUserDefinedNetworkKind } from '@utils/resources/udns/types';
+
+import { CreateProjectModalFormState } from '../types';
+
+const SelectClusterUDN: FC = () => {
+  const { t } = useNetworkingTranslation();
+
+  const { register, setValue, watch } = useFormContext<CreateProjectModalFormState>();
+  const [cudns, loaded] = useK8sWatchResource<ClusterUserDefinedNetworkKind[]>({
+    groupVersionKind: ClusterUserDefinedNetworkModelGroupVersionKind,
+    isList: true,
+  });
+
+  const selectedClusterUDN = watch('clusterUDN');
+
+  const selectedClusterUDNname = getName(selectedClusterUDN);
+
+  if (!loaded) return <InlineLoading />;
+
+  return (
+    <FormGroup fieldId="select-cluster-udn" isRequired label={t('Project name')}>
+      <Select
```

Comment:
with react-hook-form , the `register` method is meant to be used with simple uncontrolled components. for controlled components like a Select, the best practice is using the [Controller component](https://www.react-hook-form.com/api/usecontroller/controller/ )